cli/azd/pkg/ux/action_printer.go

```diff
@@ -0,0 +1,96 @@
+package ux
+
+import (
+	"context"
+)
+
+// Creates a new instance of an ActionPrinter
+func NewActionPrinter[R any]() *ActionPrinter[R] {
+	return &ActionPrinter[R]{
+		continueOnError: false,
+		initStep:        &initStep{},
```

Comment:
NewInitStep("", "") maybe?  I'm not sure which to prefer.

Reply:
Thinking about it a little more, I think I'm happy for us to keep the zero value here if we remove `NewInitStep()`